web/client/components/data/identify/DefaultViewer.jsx

```diff
@@ -97,6 +98,20 @@ class DefaultViewer extends React.Component {
         return validator.getValidResponses([response]);
     }
 
+    /**
+     * Helper method to calculate panel index properly on mobile devices.
+     */
+    getPanelIndex = (index, filterIndex = true) => {
+        // Recalculate index value of valid responses when mode is mobile
+        const { responses, validResponses } = this.getResponseProperties(this.props.isMobile);
```

Comment:
I need some clarifications, Why do we have two different behaviours in mobile and desktop?
Shouldn't both have the same results?
If I understood correctly this is happening because of the swipe interaction, is there a reason to render invalid pages on desktop? if not wouldn't be better to have same behavior in desktop and mobile so to remove always the invalid pages?

Reply:
The behavior of desktop and mobile is same. Only difference is that in desktop, the user can't swipe to look at empty render pages as the only way to interact is via the layer selector DD field. In mobile, however, the user can swipe to look at container's child which may have empty pages, so to avoid that the panel index is calculated such that user doesn't view these empty pages.